pydoctor/astbuilder.py

```diff
@@ -512,36 +485,81 @@ def _annotation_from_attrib(self, expr, ctx):
                 return _infer_type(default)
         return None
 
-    def _unstring_annotation(self, node):
+    def _annotations_from_function(self, func: ast.FunctionDef) -> Mapping[str, ast.expr]:
+        """Get annotations from a function definition.
+        @param func: The function definition's AST.
+        @return: Mapping from argument name to annotation.
+            The name C{return} is used for the return type.
+            Unannotated arguments are omitted.
+        """
+        def _get_all_args() -> Iterator[ast.arg]:
+            base_args = func.args
+            # New on Python 3.8 -- handle absence gracefully
+            try:
+                yield from base_args.posonlyargs
+            except AttributeError:
+                pass
+            yield from base_args.args
+            varargs = base_args.vararg
+            if varargs:
+                yield varargs
+            yield from base_args.kwonlyargs
+            kwargs = base_args.kwarg
+            if kwargs:
+                yield kwargs
+        def _get_all_ast_annotations() -> Iterator[Tuple[str, ast.expr]]:
+            for arg in _get_all_args():
+                if arg.annotation:
+                    yield arg.arg, arg.annotation
+            returns = func.returns
+            if returns:
+                yield 'return', returns
+        return {name: self._unstring_annotation(value)
+                for name, value in _get_all_ast_annotations()}
+
+    def _unstring_annotation(self, node: ast.expr) -> ast.expr:
         """Replace all strings in the given expression by parsed versions.
-        Returns the resulting node, or None if parsing failed.
+        @return: The unstringed node. If parsing fails, an error is logged
+            and the original node is returned.
         """
         try:
-            return _AnnotationStringParser().visit(node)
+            expr = _AnnotationStringParser().visit(node)
+            assert isinstance(expr, ast.expr), expr
+            return expr
         except SyntaxError as ex:
             self.builder.currentMod.report(
                     f'syntax error in annotation: {ex}',
                     lineno_offset=node.lineno)
-            return None
+            return node
 
 
 class _AnnotationStringParser(ast.NodeTransformer):
+    """Implementation of L{ModuleVistor._unstring_annotation()}."""
+
+    def _parse_string(self, value: str) -> ast.expr:
+        stmt, = ast.parse(value).body
+        if isinstance(stmt, ast.Expr):
```

Comment:
I think `ast.Expr` and `ast.expr` are aliases, but it would probably be better to be consistent.

Reply:
They are not the same thing: `ast.Expr` is a statement that contains an expression, while `ast.expr` is the base class for all expressions.

For example, in AST for the following code:
```py
print('hello')
f(2)
```
The second statement `s` is of type `ast.Expr` node and `s.value` contains the expression `f(2)` which is a subsclass of `ast.expr`.